R/install.snowball.R

```diff
@@ -72,69 +72,130 @@ install.snowball <- function(pkg, date, include.suggests, force.install = FALSE,
+          }
+
+        # Attempt 6, MRAN around Publish date, Archive path
+        if (is.pkg_vrs.installed(pkg.k, vrs.k) == FALSE) # if still not installed
+          {
+            # If fails try archive
+            url6 <- paste0("https://cran.microsoft.com/snapshot/", mran.published.date, "/src/contrib/", pkg_vrs.k, ".tar.gz")
+            install.packages(url6, repos = NULL, lib = lib.k, type = "source", quiet.install = quiet, dependencies = F, quiet = quiet.install)
+          }
       }
+      # install.source(pkg_vrs.k, lib.k, date, quiet.install = quiet.install)  #used to call an external function
+
 
       # 3.8 VERIFY INSTALL
       now.installed <- is.pkg_vrs.installed(pkg.k, vrs.k)
 
       # 3.8.5 If not success, try source again, forcing download of file
-      if (!now.installed) {
-        install.source(pkg_vrs.k, lib.k, date, force.download = TRUE, quiet.install = quiet.install)
-      }
+      # if (!now.installed) {
+      #  install.source(pkg_vrs.k, lib.k, date, force.download = TRUE, quiet.install = quiet.install)
+      # }
+      # 2020 05 14 Commented out as it makes failure too slow, and unlikely that tarball is incorrectly downloaded.
 
       # 3.8.6 Verify install again
-      now.installed <- is.pkg_vrs.installed(pkg.k, vrs.k)
+      # now.installed <- is.pkg_vrs.installed(pkg.k, vrs.k)
 
       # 3.9 Installation failed
       if (!now.installed) {
 
         # If using different R version, attribute to that:
         rv <- r.version.check(date)
         if ((rv$r.using.minor != rv$r.need.minor) | (rv$r.using.major != rv$r.need.major)) {
+          message1() # to skip a line
           message2()
+          message("Installation failed!")
           message1(
-            "As mentioned before, there is a discrepancy between the R Version you are using R-", rv$r.using.full,
-            "\nand the one that matches the date (", date, ") you entered: R-", rv$r.need.full, ". ",
-            "It was worth a try \nbut the install did not work. Follow the instructions below to",
-            " try this script in the correct R version."
+            "A likely culprit is that there is a discrepancy between the R Version you are using R-", rv$r.using.full,
+            "\nand the one that matches the date (", date, ") you entered: R-", rv$r.need.full, ".\n",
+            "Many packages cannot be installed in newer versions of R.\n",
+            "Either use a newer version of the package, or follow the instructions below to run the older version of R."
           )
 
 
-          # Long instructions written above
+          # Instructions for switching R to specific date (separate function)
           msg.R.switch(date)
         }
 
-        # PENDINGrename back the flders taht were renamed
-        #
+        # R TOOLS CHECK
+        # WINDOWS and missing R Tools
+        if (Sys.info()["sysname"] == "Windows" & Sys.which("make") == "") {
+          # Draft message
+          msg <- paste0(
```

Comment:
`message()`, `message1()` and `message2()` already concatenate strings by default, no need for `paste()` here.

Note to self: change this